packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -104,7 +104,7 @@ module.exports = function watchmanCrawl(
 
             clocks[root] = response.clock;
             response.files.forEach(fileData => {
-              const name = root + path.sep + fileData.name;
+              const name = path.normalize(path.join(root, fileData.name));
```

Comment:
Oh, got it. Is it also for performance that `path.join` wasn't used?

The alternative would be to use a RegExp to replace all `/` with `path.sep`. Do you think that'd be reasonnable? It sounds like the performance would be better because it wouldn't have to do any parsing that `normalize` is probably doing.